File: gdchat/服务端/comm-modules/im-core-module/src/main/java/com/shiku/im/entity/PayConfig.java
package com.shiku.im.entity;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 *
 * @version V1.0
 * @Description: TODO(todo)
 * @date 2020/1/4 9:43
 */
@Data
@Document(value="payConfig")
public class PayConfig {

    private @Id long id=10000;

    /**
     * 单笔最大转账金额
     */
    private double maxTransferAmount;

    /**
     * 单笔最大红包金额
     */
    private double maxRedpacktAmount=200.0;

    /**
     * 单笔群红包最大发送个数
     */
    private int maxRedpacktNumber = 10;

    /**
     * 单笔最大提现金额
     */
    private double maxWithdrawAmount;

    /**
     * 单笔最大充值金额
     */
    private double maxRechargeAmount;

    /**
     * 单笔付款码支付最大金额
     */
    private double maxCodePaymentAmount;

    /**
     * 单日付款码支付最大金额
     */
    private double dayMaxCodePayAmount;

    /**
     * 单笔二维码收款最大金额
     */
    private double maxCodeReceiptAmount;

    /**
     * 单日二维码收款最大金额
     */
    private double dayMaxCodeReceiptAmount;

    /**
     * 用户单日转账最大金额
     */
    private double dayMaxTransferAmount;

    /**
     * 用户单日红包最大金额
     */
    private double dayMaxRedpacktAmount;

    /**
     * 用户单日提现最大金额
     */
    private double dayMaxWithdrawAmount;

    /**
     * 用户单日充值最大金额
     */
    private double dayMaxRechargeAmount;

    /**
     * 零钱提现费率
     */
    private double myChangeWithdrawRate = 0.006;

    /**
     * 是否开启扫码手动充值
     */
    private byte isOpenManualPay = 0;

    /**
     * 是否开启云钱包 1:开启 0:关闭
     */
    private byte isOpenCloudWallet = 0;

    /**
     * 云钱包开户用户是否默认不冻结
     */
    private byte isDefaultFreeze = 0;

    /**
     * 云钱包新开户充值多少小时后可交易
     */
    private double newYopUserRechargeTime = 24;

    /**
     * 云钱包新开户第一笔充值限额
     */
    private double newYopUserFirstRecharge = 10;

    /**
     * 后台管理中单笔最大充值金额
     */
    private double consoleMaxRechargeAmount = 1;

    /**
     * 后台管理中单笔最大手工提现金额
     */
    private double consoleMaxCodePaymentAmount = 1;

}

File: gdchat/服务端/parent/mianshi-im-api/src/main/webapp/pages/console/js/payConfig.js
layui.use(['form','jquery',"layer"],function() {
    var form = layui.form,
        $ = layui.jquery,
        layer = parent.layer === undefined ? layui.layer : top.layer;

    //非管理员登录屏蔽操作按钮
    if(localStorage.getItem("IS_ADMIN")==0){
        $(".save").remove();
    }

    //获取当前系统支付配置
    Common.invoke({
        url : request('/console/payConfig'),
        data : {},
        successMsg : false,
        errorMsg : "获取数据失败,请检查网络",
        success : function(result) {
            fillParameter(result.data);
        },
        error : function(result) {
        }

    });

    form.on("submit(walletConfig)",function(){
        var payConfig = {};
        payConfig.id = 10000;
        payConfig.maxTransferAmount = $(".maxTransferAmount").val();
        payConfig.maxRedpacktAmount = $(".maxRedpacktAmount").val();
        payConfig.maxRedpacktNumber = $(".maxRedpacktNumber").val();
        payConfig.maxWithdrawAmount = $(".maxWithdrawAmount").val();
        payConfig.maxRechargeAmount = $(".maxRechargeAmount").val();
        payConfig.maxCodePaymentAmount = $(".maxCodePaymentAmount").val();
        payConfig.maxCodeReceiptAmount = $(".maxCodeReceiptAmount").val();
        payConfig.dayMaxTransferAmount = $(".dayMaxTransferAmount").val();
        payConfig.dayMaxRedpacktAmount = $(".dayMaxRedpacktAmount").val();
        payConfig.dayMaxWithdrawAmount = $(".dayMaxWithdrawAmount").val();
        payConfig.dayMaxRechargeAmount = $(".dayMaxRechargeAmount").val();
        payConfig.dayMaxCodePayAmount = $(".dayMaxCodePayAmount").val();
        payConfig.dayMaxCodeReceiptAmount = $(".dayMaxCodeReceiptAmount").val();
        payConfig.isOpenManualPay = $(".isOpenManualPay").val();
        payConfig.isOpenCloudWallet = $(".isOpenCloudWallet").val();
        payConfig.isDefaultFreeze = $(".isDefaultFreeze").val();
        payConfig.myChangeWithdrawRate = $(".myChangeWithdrawRate").val();
        payConfig.newYopUserRechargeTime = $(".newYopUserRechargeTime").val();
        payConfig.newYopUserFirstRecharge = $(".newYopUserFirstRecharge").val();
        payConfig.consoleMaxRechargeAmount = $(".consoleMaxRechargeAmount").val();
        payConfig.consoleMaxCodePaymentAmount = $(".consoleMaxCodePaymentAmount").val();

        Common.invoke({
            url : request('/console/payConfig/set'),
            data : payConfig,
            successMsg : "系统配置修改成功",
            errorMsg : "修改系统配置失败,请检查网络",
            success : function(result) {

            },
            error : function(result) {

            }

        });
        return false;
    })

});

function fillParameter(data){
    //判断字段数据是否存在
    function nullData(data){
        if(data == '' || data == "undefined" || data==null){
            return "";
        }else{
            return data;
        }
    }
    // 数据回现
    $(".maxTransferAmount").val(data.maxTransferAmount);
    $(".maxRedpacktAmount").val(data.maxRedpacktAmount);
    $(".maxRedpacktNumber").val(data.maxRedpacktNumber);
    $(".maxWithdrawAmount").val(data.maxWithdrawAmount);
    $(".maxRechargeAmount").val(data.maxRechargeAmount);
    $(".maxCodePaymentAmount").val(data.maxCodePaymentAmount);
    $(".maxCodeReceiptAmount").val(data.maxCodeReceiptAmount);
    $(".dayMaxTransferAmount").val(data.dayMaxTransferAmount);
    $(".dayMaxRedpacktAmount").val(data.dayMaxRedpacktAmount);
    $(".dayMaxWithdrawAmount").val(data.dayMaxWithdrawAmount);
    $(".dayMaxRechargeAmount").val(data.dayMaxRechargeAmount);
    $(".dayMaxCodePayAmount").val(data.dayMaxCodePayAmount);
    $(".dayMaxCodeReceiptAmount").val(data.dayMaxCodeReceiptAmount);
    $(".isOpenManualPay").val(data.isOpenManualPay);
    $(".isOpenCloudWallet").val(data.isOpenCloudWallet);
    $(".isDefaultFreeze").val(data.isDefaultFreeze);
    $(".myChangeWithdrawRate").val(data.myChangeWithdrawRate);
    $(".newYopUserRechargeTime").val(data.newYopUserRechargeTime);
    $(".newYopUserFirstRecharge").val(data.newYopUserFirstRecharge);
    $(".consoleMaxRechargeAmount").val(data.consoleMaxRechargeAmount);
    $(".consoleMaxCodePaymentAmount").val(data.consoleMaxCodePaymentAmount);
    //重新渲染
    layui.form.render();
}